meta-mbl-distro/recipes-core/mbl-initramfs-init/files/initramfs-init-script.sh

```diff
@@ -64,20 +70,24 @@ echo "Booting from init script in initramfs"
 # Workaround findfs failure on Raspberry Pi 3: unable to resolve 'LABEL=rootfs1'
 sleep 0.1
 
-BOOTFLAG_PARTITION="$(findfs LABEL=bootflags)"
-mkdir -p /mnt/bootflags
-mount $BOOTFLAG_PARTITION /mnt/bootflags
+# The boot flags (which indicate which rootfs bank to use) are temporarily stored
+# in the log partition. Mount the log partition so we can check them.
+LOG_PARTITION="$(findfs "LABEL=${LOG_PARTITION_LABEL}")"
+mkdir -p "$LOG_MOUNT_POINT"
+mount "$LOG_PARTITION" "$LOG_MOUNT_POINT"
 
-# Check for the existence of a "rootfs2" on BOOTFLAG_PARTITION.
-# If it exists - use findfs to find the file system with label "rootfs2"
-# Else - use findfs to find the file system with ROOTFS_LABEL
-ROOTFS_LABEL="rootfs2"
-ROOTFS_FILE_INDICATOR="/mnt/bootflags/rootfs2"
-if [ ! -f "$ROOTFS_FILE_INDICATOR" ]
+# Check for the existence of a flag file indicating that we should use the
+# second rootfs bank rather than the first.
+ROOTFS_LABEL="${ROOTFS_LABEL_BASE}2"
+if [ ! -f "${BOOTFLAGS_DIR}/${ROOTFS_LABEL}" ]
 then
-    ROOTFS_LABEL="rootfs1"
+    ROOTFS_LABEL="${ROOTFS_LABEL_BASE}1"
 fi
 
+# We'll mount the log partition again after we've done the switch_root so don't
+# leave it hanging around here.
+umount "$LOG_PARTITION"
```

Comment:
I'm not sure if I agree with this operation. Systemd is smart enough to mount what needs to be mounted.
Unless we identify a potential security risk by leaving it mounted.

And I know that it is not part of this PR, but maybe we could think about logging initramfs errors into the log partition.

Reply:
I will make the comment more clear... I didn't add the `umount` just because we'll try to mount it again later, I added the `umount` because we'll try to mount it again later _at a different directory_.

We mount it here at `$MBL_LOG_MOUNT_POINT`, then we do the `switch_root`, then systemd will try to mount it again at `$MBL_LOG_MOUNT_POINT`, but after the `switch_root`, the path `$MBL_LOG_MOUNT_POINT` does not refer to the same directory as it did before.
```
root_before_switch_root
 |-var
    |-log ($MBL_LOG_MOUNT_POINT before switch_root)
 |-mnt
    |-rootfs (root_after_switch_root)
       |-var
          |-log ($MBL_LOG_MOUNT_POINT after switch_root)
```

It would be nice to just mount the log partition at `/mnt/rootfs/var/log` in the initramfs script and leave it mounted, but we need to mount the log partition to check the boot flags before we can mount the proper rootfs at `/mnt/rootfs`.

Reply:
I suppose we could potentially do
```
mount --move "${MBL_LOG_MOUNT_POINT}" "/mnt/root${MBL_LOG_MOUNT_POINT}"
```

Reply:
Right. In this case, you can use `mount --move /var/log /mnt/rootfs/var/log` like we do for `/dev/`, `/proc` and `/sys`.

Reply:
I added my previous comment without refreshing the page. I'd prefer to use the `mount --move` instead of unmounting and then mounting again with systemd.

Reply:
https://github.com/ARMmbed/meta-mbl/pull/641/commits/474b0cdef7d933b77c7e70226c3ab77c52c4ff39